File: atcoder/abc130B_bounding.ml
(* Vicfred
 * https://atcoder.jp/contests/abc130/tasks/abc130_b
 * implementation
 * *)
Scanf.scanf "%d %d\n" @@ fun n x ->
    Array.init n (fun _ -> Scanf.scanf " %d" ( + ) 0)
    |> Array.fold_left (fun (a, b) c -> (if b + c <= x then a + 1 else a), b + c) (1, 0)
    |> fst
    |> Printf.printf "%d\n"

File: atcoder/abc130B_bounding.nim
# unihernandez22
# https://atcoder.jp/contests/abc130/tasks/abc130_b
# simulation

import strutils, sequtils

var n, x: int
(n, x) = stdin.readLine.split.map(parseInt)

let l = stdin.readLine.split.map(parseInt)

var
  d = newSeq[int](n+1)
  ans: int

for i in 1 .. n:
  d[i] = d[i-1] + l[i-1]
  if d[i] > x: break
  ans = i

echo ans + 1
